xformers/ops/memory_efficient_attention.py

```diff
@@ -837,6 +867,7 @@ def memory_efficient_attention(
     value: torch.Tensor,
     attn_bias: Optional[Union[torch.Tensor, AttentionMask]] = None,
     p: float = 0.0,
+    has_custom_scale: bool = False,
```

Comment:
Can we rename `has_custom_scale: bool` to `scale: Optional[float]` everywhere?
This would allow to have any scale, not just either 1 or `q.shape[-1] ** -0.5`.

We can keep `has_custom_scale` in `AttentionOpDispatch` tho

Reply:
I was thinking about this too, but I'm not sure whether this can be that flexible to all kernels. I should get some senses after supporting the CUTLASS one.

Reply:
Yes it should be possible for the CUTLASS kernel as well (worst case scenario we just disable this kernel when using custom scale, which is the case with the PR at the moment I think)